src/NUnitFramework/framework/Internal/ArrayHelper.cs

```diff
@@ -0,0 +1,31 @@
+// Copyright (c) Charlie Poole, Rob Prouse and Contributors. MIT License - see LICENSE.txt
+
+#nullable enable
+
+using System;
+using System.Runtime.CompilerServices;
+
+namespace NUnit.Framework.Internal
+{
+    internal static class ArrayHelper
+    {
+#if NET45_OR_GREATER
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
```

Comment:
Only comment from me. Would it make any difference to have `[MethodImpl(256)]` as in https://github.com/nunit/nunit/pull/4035/files#diff-a8ef56950756871bf762f8132c715423e246747709de814a3734d7bb00377448 ? or does that make any difference ?

Reply:
Thanks @mikkelbu To be honest, I'm not sure about difference.

`[MethodImpl(256)]` and `MethodImpl(MethodImplOptions.AggressiveInlining)` are semantically identical since `MethodImplOptions.AggressiveInlining = 256`, with the exception that using 256 explicitly allows the code to compile on build targets which predate the enum value (introduced in .NET 4.5). Though the JIT still needs to support reading the flag from the IL for it to make a difference, and I'm not sure when that was added.

If older JITs (ex: .NET 3.5 / CLR 2) don't recognize the flag then the main difference I could see it making is if a .NET 4.5+ app were to reference a .NET 3.5 or .NET4 build of the framework. Which is possible, though I'd think it rare given nuget "should" likely serve the .NET4.5 build of the framework to the app when restored. In that case, I think the JIT should still inline it even if we don't explicitly tell it because it is a small, simple method non-polymorphic method (a few of the heuristics are listed [here](https://devblogs.microsoft.com/premier-developer/a-common-execution-path-optimization/)).

Happy to make the change though if we think it could help keep the code cleaner or more readable. I may put it behind a named constant if we do use it in multiple places in the codebase. I should be able to get to coding it later tonight. 😃

Reply:
Then I think we should merge it as it is :)

Reply:
Thanks @mikkelbu !